src/color.rs

```diff
@@ -126,6 +154,171 @@ pub struct Color {
+    /// let actual = Color::rgba(35.0, 200.0, 180.0, 0.5);
+    /// assert_eq!(expected, actual);
+    /// ```
+    /// 
+    /// Values that are outside the accepted RGB or alpha range will result in a panic
+    /// 
+    /// ```should_panic
+    /// use turtle::Color;
+    /// // This will not work as 1.1 is greater than the maximum allowed value for alpha
+    /// let color = Color::rgba(255.0, 255.0, 255.0, 1.1);
+    /// ```
+    /// [`Color`]: ./index.html
+    /// [`RGB`]: https://developer.mozilla.org/en-US/docs/Glossary/RGB
+    pub fn rgba(red: f64, green: f64, blue: f64, alpha: f64) -> Self {
+        let color = Color { red, green, blue, alpha };
+        assert!(color.is_valid(), "Invalid color: {:?}. See the color module documentation for more information.", color);
+        color
+    }
+
+    /// Create a new `Color` from the given [`HSL`] values with alpha set to 1.0.
+    /// 
+    /// The expected value ranges are:
+    /// 
+    /// * 0.0 &le; `hue` &le; 360.0
+    /// * 0.0 &le; `saturation` &le; 1.0
+    /// * 0.0 &le; `lightness` &le; 1.0
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// let black = Color::from("black");
+    /// let black_hsl = Color::hsl(0.0, 0.0, 0.0);
+    /// assert_eq!(black, black_hsl);
+    /// 
+    /// let white = Color::from("white");
+    /// let white_hsl = Color::hsl(0.0, 1.0, 1.0);
+    /// assert_eq!(white, white_hsl);
+    /// ```
+    /// [`HSL`]: https://en.wikipedia.org/wiki/HSL_and_HSV
+    pub fn hsl(hue: f64, saturation: f64, lightness: f64) -> Self {
+        Color::hsla(hue, saturation, lightness, 1.0)
+    }
+
+    /// Create a new `Color` from the given [`HSL`] values and the given alpha value.
+    /// 
+    /// The expected value ranges are:
+    /// 
+    /// * 0.0 &le; `hue` &le; 360.0
+    /// * 0.0 &le; `saturation` &le; 1.0
+    /// * 0.0 &le; `lightness` &le; 1.0
+    /// * 0.0 &le; `alpha` &le; 1.0
+    /// 
+    /// ```rust
+    /// use turtle::Color;
+    /// let black = Color::from("black").with_alpha(0.5);
+    /// let black_hsla = Color::hsla(0.0, 0.0, 0.0, 0.5);
+    /// assert_eq!(black, black_hsla);
+    /// 
+    /// let white = Color::from("white").with_alpha(0.75);
+    /// let white_hsla = Color::hsla(0.0, 1.0, 1.0, 0.75);
+    /// assert_eq!(white, white_hsla);
+    /// ```
+    /// [`HSL`]: https://en.wikipedia.org/wiki/HSL_and_HSV
+    pub fn hsla(hue: f64, saturation: f64, lightness: f64, alpha: f64) -> Self {
+        if hue < 0. || hue > 360. {
+            panic!("{} is not a valid value for hue, values must be between 0.0 and 360.0");
+        }
+
+        if saturation < 0. || saturation > 1. {
+            panic!("{} is not a valid value for saturation, values must be between 0.0 and 1.0");
+        }
+
+        if lightness < 0. || lightness > 1. {
+            panic!("{} is not a valid value for lightness, values must be between 0.0 and 1.0");
+        }
```

Comment:
Sure, I was just delegating it to the existing is_valid() check however as it is a common message we're already using for other color creation.

Reply:
Yes. That is a good thought. I really like how specific these error messages are though and I think it would be great to have one for alpha as well. These messages are actually probably much more helpful than the other ones we've had so far. :smile:

Reply:
Sure, I can also do the same in rgba constructor